extensions/api/v1alpha1/sandboxclaim_types.go

```diff
@@ -37,6 +21,43 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 // Important: Run "make" to regenerate code after modifying this file
 
+const (
+	// ClaimExpiredReason is the reason used in conditions/events when a claim expires.
+	ClaimExpiredReason = "ClaimExpired"
+)
+
+// ShutdownPolicy describes the policy for shutting down the underlying Sandbox when the SandboxClaim expires.
+// +kubebuilder:validation:Enum=Delete;Retain
+type ShutdownPolicy string
+
+const (
+	// ShutdownPolicyDelete deletes the SandboxClaim (and cascadingly the Sandbox) when expired.
+	ShutdownPolicyDelete ShutdownPolicy = "Delete"
+
+	// ShutdownPolicyRetain keeps the SandboxClaim when expired (Status will show Expired).
+	// The underlying SandboxClaim resources (Pod, Service) are deleted to save resources,
+	// but the SandboxClaim object itself remains.
+	ShutdownPolicyRetain ShutdownPolicy = "Retain"
+)
+
+// Lifecycle defines the lifecycle management for the SandboxClaim.
+type Lifecycle struct {
+	// ShutdownTime is the absolute time when the SandboxClaim expires.
+	// This time governs the lifecycle of the claim. It is propagated to the
+	// underlying Sandbox, overriding any expiration logic defined there.
```

Comment:
Actually, do we need to propagate a claim's shutdown time to its sandbox? The sandbox will be deleted post claim's shutdown time, even if the sandbox's shutdown time is scheduled later. However, if the sandbox is scheduled to be deleted before the claim, trying to propagate the claim's shutdown time to the sandbox to extend it seems misleading, given that the lifecycle is about the sandbox itself. If we want propagation, it should use a separate field.

Reply:
This is a good point. Yes, i agree that propagating the time adds unnecessary complexity (race conditions, unsetting values) for little gain.

I will remove the propagation logic entirely, so that the SandboxClaim controller acts purely as an orchestrator:

- If the claim's ShutdownTime is reached -> Enforce Policy (Delete Claim or Mark Expired).
- We don't touch `Sandbox.Spec.ShutdownTime`.

This treats the Claim as a higher-order constraint that terminates the Sandbox regardless of the Sandbox's internal configuration. This should simplify the controller significantly. Sorry i was overcomplicating a lot.